### ISSUE TITLE
🐛[BUG]: FusedStage masked updates can change inactive systems' cells

### ISSUE THREAD

**Issue by architdatar** [NVIDIA/nvalchemi-toolkit#115]
### Version

v0.1.0 commit 01c99d5cde6f63d6f662b071a9f408d3bfc12b0a

### On which installation method(s) does this occur?

Source

### Describe the issue

`FusedStage` runs multiple sub-stages on one shared `Batch`. Each sub-stage should update only systems whose `status` matches that stage. For example, in a batch of 2 systems, one system may be in a fixed-cell NVE/NVT validation stage while another system in the same live batch is in a variable-cell FIRE/NPT-style stage.

The masked update helpers restore inactive systems' `positions` and `velocities` after a sub-stage runs, but they do not restore `cell`. As a result, a cell-mutating stage can change every system's cell, including systems that are inactive for that stage.

Expected behavior: inactive systems' `cell` values should be protected during masked updates, just like their positions and velocities.

Actual behavior: inactive systems' `cell` values can change when another system is updated by a cell-mutating stage.

### Minimum reproducible example

```python
from collections import OrderedDict

import torch
from torch import nn

from nvalchemi.data import AtomicData, Batch
from nvalchemi.dynamics import FIREVariableCell
from nvalchemi.models.base import BaseModelMixin, ModelConfig


class ConstantStressModel(nn.Module, BaseModelMixin):
    """Minimal model that provides forces and stress for variable-cell dynamics."""

    def __init__(self) -> None:
        super().__init__()
        self.model_config = ModelConfig(outputs=frozenset({"energy", "forces", "stress"}))

    @property
    def embedding_shapes(self) -> dict[str, tuple[int, ...]]:
        return {}

    def compute_embeddings(self, data: AtomicData, **kwargs: object) -> AtomicData:
        return data

    def forward(self, batch: Batch) -> OrderedDict[str, torch.Tensor]:
        return OrderedDict(
            energy=torch.zeros(batch.num_graphs, 1, device=batch.device),
            forces=torch.zeros_like(batch.positions),
            stress=torch.zeros(batch.num_graphs, 3, 3, device=batch.device),
        )


def make_data(offset: float) -> AtomicData:
    data = AtomicData(
        atomic_numbers=torch.tensor([1], dtype=torch.long),
        positions=torch.tensor([[offset, 0.0, 0.0]], dtype=torch.float32),
        cell=torch.eye(3).unsqueeze(0) * (offset + 1.0),
        pbc=torch.tensor([[True, True, True]]),
    )
    data.forces = torch.zeros(1, 3)
    data.energy = torch.zeros(1, 1)
    data.add_node_property("velocities", torch.zeros(1, 3))
    return data


def main() -> None:
    batch = Batch.from_data_list([make_data(0.0), make_data(1.0)])
    dynamics = FIREVariableCell(model=ConstantStressModel(), dt=1.0, n_steps=1)
    dynamics._ensure_state_initialized(batch)
    dynamics._state.cell_velocity.fill_(0.05)

    mask = torch.tensor([True, False], dtype=torch.bool)
    inactive_before = batch.cell[1].clone()

    # Pre-update is called in FusedStage _step_impl 
    dynamics._masked_pre_update(batch, mask)

    inactive_after = batch.cell[1].clone()
    print(f"inactive cell before:\n{inactive_before}")
    print(f"inactive cell after masked pre_update:\n{inactive_after}")

    if not torch.equal(inactive_before, inactive_after):
        print()
        print("BUG REPRODUCED: inactive system cell changed during masked update.")
        return


if __name__ == "__main__":
    main()
```

### Relevant log output

```shell
inactive cell before:
tensor([[2., 0., 0.],
        [0., 2., 0.],
        [0., 0., 2.]])
inactive cell after masked pre_update:
tensor([[2.1000, 0.1000, 0.1000],
        [0.1000, 2.1000, 0.1000],
        [0.1000, 0.1000, 2.1000]])

BUG REPRODUCED: inactive system cell changed during masked update.
```

### Environment details

```shell
+ Environment location: Source checkout
+ OS: Linux
+ Python/package manager: uv
+ GPU: not required for this reproducer
```
